dashboard/test/models/user_test.rb

```diff
@@ -142,6 +142,21 @@ class UserTest < ActiveSupport::TestCase
     assert_equal user.user_school_infos.where(end_date: nil).count, 1
   end
 
+  test 'update_school_info with specific school overwrites user school info for user without email' do
+    user = create :teacher, :with_school_info
+    new_school_info = create :school_info
+
+    user.email = ""
+    user.save(validate: false)
+
```

Comment:
```suggestion
    refute user.valid?

```
One more assertion might help call out that we're testing an unusual case here.  It might also be worth a comment in this test explaining why we chose to test against an invalid user record.

Reply:
Done and added a comment to this test.